### PR TITLE
[Stripe, VCR] Updates credit_card_cloner_spec.rb

### PR DESCRIPTION
**Pull request by filipefurtad0**
Replaces stubs on `./spec/lib/stripe/credit_card_cloner_spec.rb` with real VCR requests.

To my understanding, this tests both the files:
- `./lib/stripe/credit_card_cloner.rb`
- `./lib/stripe/credit_card_clone_finder.rb`

#### What? Why?

- Partially addresses  #12001

#### What should we test?

- green build

#### Release notes

Changelog Category (reviewers may add a label for the release notes):

- [ ] User facing changes
- [ ] API changes (V0, V1, DFC or Webhook)
- [x] Technical changes only
- [ ] Feature toggled

The title of the pull request will be included in the release notes.

### CODE REVIEW

**Comment by filipefurtad0** [comment]
I'm confident this is working as expected (:crossed_fingers: ).

However, I'm not really sure how to improve assertions in this case. I think it would be great to demonstrate that:
- test case 1: `customer` remains `nil`; `payment_method_id` changes from pm_* to `new_payment_method` (pm_*)
- test case 2: `customer` changes from `nil` to `cus_*`; `payment_method_id` changes from pm_* to `new_payment_method` (pm_*)

I can observe that this is the case and that we're good, if I run the spec in the console :heavy_check_mark: 

The variables above can in principle be retrieved from the VCR cassettes; but I'm not sure what the best way to do this is. I'm moving to code review, as I believe this is really the final missing bit, and perhaps @openfoodfoundation/core-devs have a good suggestion. If not, happy to investigate further.

File: spec/lib/stripe/credit_card_cloner_spec.rb [hunk 1]
```diff
@@ -5,62 +5,62 @@
...
-            .to_return(payment_method_response_mock)
-        end
+        let!(:payment_method_id) { pm_card.id }
 
         it "clones the payment method only" do
-          customer_id, payment_method_id = cloner.find_or_clone
+          customer_id, new_payment_method_id = cloner.find_or_clone
 
-          expect(payment_method_id).to eq new_payment_method_id
+          expect(payment_method_id).not_to eq new_payment_method_id
```
Comment (rioug) [inline spec/lib/stripe/credit_card_cloner_spec.rb:60]: I wondering why the assertion changed here ? was the test before wrong ?

File: spec/lib/stripe/credit_card_cloner_spec.rb [hunk 2]
```diff
@@ -5,76 +5,83 @@
...
+    let!(:user) { create(:user, email: "apple.customer@example.com") }
+    let!(:enterprise) { create(:enterprise) }
 
-      let(:credit_card) { create(:credit_card, user: create(:user)) }
-      let(:stripe_account_id) { "abc123" }
+    let(:secret) { ENV.fetch('STRIPE_SECRET_TEST_API_KEY', nil) }
 
-      let(:cloner) { Stripe::CreditCardCloner.new(credit_card, stripe_account_id) }
+    describe "#find_or_clone", :vcr, :stripe_version do
+      include StripeStubs
```
Comment (rioug) [inline spec/lib/stripe/credit_card_cloner_spec.rb:14]: Do we still need this ? I though we were removing stubs here.

**Review by rioug — APPROVED** [review]
Looks good to me ! Just a couple of comments.

File: spec/lib/stripe/credit_card_cloner_spec.rb (hunk 1 above)
Comment (filipefurtad0) [reply spec/lib/stripe/credit_card_cloner_spec.rb:60]: I don't think it was wrong, but when using the stubs `new_payment_method_id` can be defined in advance and `payment_method_id` is changed by `cloner.find_or_clone`.

With the real HTTP calls, I found it more appropriate not to change `payment_method_id` and compare with `new_payment_method_id`, which is created when cloning, in this case.

File: spec/lib/stripe/credit_card_cloner_spec.rb (hunk 2 above)
Comment (filipefurtad0) [reply spec/lib/stripe/credit_card_cloner_spec.rb:14]: Indeed we don't - left forgotten. Thank you.

File: spec/lib/stripe/credit_card_cloner_spec.rb [hunk 3]
```diff
@@ -5,62 +5,62 @@
...
+      let(:credit_card) { create(:credit_card, gateway_payment_profile_id: pm_card.id, user:) }
+
+      let(:pm_card) {
+        Stripe::PaymentMethod.create(
+          {
+            type: 'card',
+            card: {
+              number: '4242424242424242',
+              exp_month: 8,
+              exp_year: 2026,
```
Comment (mkllnk) [inline spec/lib/stripe/credit_card_cloner_spec.rb:33]: We'll need to update that in a couple of years.

**Review by mkllnk — APPROVED** [review]
Great. I think that those assertions are good enough. We can't predict the customer ids generated by Stripe and we don't really care. We just want to know that it changes.

File: spec/lib/stripe/credit_card_cloner_spec.rb (hunk 3 above)
Comment (filipefurtad0) [reply spec/lib/stripe/credit_card_cloner_spec.rb:33]: Ahh, yes I forgot the usual bit of setting a future date... I'll keep this in mind and change it in the upcoming PR.

**Pull request merged**